### PR TITLE
feat(instrumentation-lambda): Flush MeterProvider at end of handler

### PR DESCRIPTION
**Pull request by Aneurysm9**
Signed-off-by: Anthony J Mirabella <a9@aneurysm9.com>

## Which problem is this PR solving?

The Lambda instrumentation wrapper does not flush metrics at the end of handler invocation.

## Short description of the changes

Flushes the `MeterProvider` at the end of handler invocation.

### CODE REVIEW

**Review by pichlermarc — APPROVED** [review]
Thanks for adding this :slightly_smiling_face: Looks good to me. :slightly_smiling_face: 
Failing tests look unrelated to these changes, I think a rebase should fix those. :)

File: plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts [hunk 1]
```diff
@@ -294,6 +314,17 @@ export class AwsLambdaInstrumentation extends InstrumentationBase {
       );
       callback();
     }
+    if (this._metricForceFlusher) {
+      this._metricForceFlusher().then(
+        () => callback(),
```
Comment (legendecas) [inline plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts:319]: Would you mind adding a test to verify that this callback is only invoked once when the tracer and meter are both registered?

Comment (Aneurysm9) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts:319]: good catch!  I've added a test and it's failing.  Working on a fix now.

```
1) force flush
       should callback once after force flush providers:

      AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

2 !== 1

      + expected - actual

      -2
      +1
      
      at Context.<anonymous> (test/integrations/lambda-handler.force-flush.test.ts:236:12)
```

Comment (pichlermarc) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts:319]: Good catch, thanks for following up on this!

Comment (Aneurysm9) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/src/instrumentation.ts:319]: https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1384 should fix this.

**Pull request merged**